sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,155 @@
+from azure.keyvault.secrets import SecretClient
+from ._settingselector import SettingSelector
+from ._user_agent import USER_AGENT
+
+
+class AzureAppConfigurationProvider:
+
+    def __init__(self):
+        self._dict = {}
+        self._trim_prefixes = []
+        self._client = None
+
+    @classmethod
+    def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
+        """
+        Requires either a connection-string, or an Endpoint with a Credential. Loads the selected configuration
+         settings into itself for usage.
+        Optional parameters:
+        selectors (List of SettingSelector for selecting which applicationconfiguration settings to load),. If not
+         specified, all key-values with the empty label will be loaded.
+        trimmed_key_prefixes (remove prefixes in key name, list of what to trim),
+        key_vault_options (Configurations for connecting to Key Vault(s))
+        """
+        provider = AzureAppConfigurationProvider()
+
+        key_vault_options = kwargs.pop("key_vault_options", None)
+
+        provider.buildprovider(connection_string, endpoint,
+                               credential, key_vault_options)
+
+        selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
+        provider._trim_prefixes = kwargs.pop("trimmed_key_prefixes", [])
+
+        provider._dict = {}
+        secret_clients = {}
+
+        for select in selects:
+            configurations = provider._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter)
+            for config in configurations:
+                if config.content_type is None:
+                    # Deals with possible null value via Rest API
+                    provider._dict[provider.trim(config.key)] = config.value
+                elif config.content_type == "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8":
+                    provider.resolve_keyvault_references(
+                        config, key_vault_options, secret_clients)
+                elif "application/json" in config.content_type:
+                    j_object = json.loads(config.value)
+                    provider._dict[provider.trim(config.key)] = j_object
+                else:
+                    provider._dict[provider.trim(config.key)] = config.value
+        return provider
+
+    def buildprovider(self, connection_string, endpoint, credential, key_vault_options):
+        usesKeyVault = False
+
+        if (key_vault_options is not None and
+                (key_vault_options.credential is not None or key_vault_options.secret_clients is not None or
+                 key_vault_options.secret_resolver is not None)):
+            usesKeyVault = True
+
+        headers = {}
+        correlation_context = "RequestType=Startup"
+
+        if usesKeyVault:
+            correlation_context += ",UsesKeyVault"
```

Comment:
why not move this inside the above if-block and get rid of the usesKeyVault flag

Reply:
I am not sure why I did it this way. Good catch.